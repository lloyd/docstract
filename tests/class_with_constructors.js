/**
 * A test of basic class parsing
 */

/**
 * @class URL
 * A class which parses a url and exposes its various
 * components separately.
 */

/**
 * @constructor
 * This is the first constructor of the class
 * @param {foo} bar it accepts a bar argument of type foo
 */


/**
 * @constructor
 * This is the second (no-arg) constructor of the class
 */

/** @endclass */
