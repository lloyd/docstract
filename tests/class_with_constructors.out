{
  "classes": [
    {
      "constructors": [
        {
          "desc": "This is the first constructor of the class", 
          "params": [
            {
              "desc": "it accepts a bar argument of type foo", 
              "name": "bar", 
              "type": "foo"
            }
          ], 
          "source_lines": [
            11, 
            17
          ]
        },
        {
          "desc": "This is the second (no-arg) constructor of the class", 
          "source_lines": [
            18,
            22
          ]
        }
      ],
      "desc": "A class which parses a url and exposes its various\ncomponents separately.", 
      "name": "URL", 
      "source_lines": [
        5, 
        10
      ]
    }
  ], 
  "desc": "A test of basic class parsing", 
  "filename": "tests/class_with_constructors.js", 
  "module": "class_with_constructors", 
  "source_lines": [
    1, 
    4
  ]
}
